compiler/src/main/java/motif/compiler/ir/CompilerClass.kt

```diff
@@ -43,8 +45,11 @@ class CompilerClass(
     }
 
     override val methods: List<IrMethod> by lazy {
-        MoreElements.getLocalAndInheritedMethods(typeElement, env.typeUtils, env.elementUtils)
+        val methods: List<ExecutableElement> = MoreElements.getLocalAndInheritedMethods(typeElement, env.typeUtils, env.elementUtils)
                 .filter { it.enclosingElement.toString() != "java.lang.Object" }
+        val nonPrivateStaticMethods: List<ExecutableElement> = ElementFilter.methodsIn(typeElement.enclosedElements)
```

Comment:
Its debatable whether presence of private methods should be reporting an error instead of silently ignoring them. Since we don't use annotations in the Motif api for the Provider methods it might be justifiable to report. In practice I have never seen usage of private methods in Modules before though technically they might have a purpose as a decomposition tool.

Reply:
> they might have a purpose as a decomposition tool.

Yeah this is the use case I was thinking of. I can see both arguments here but I tend to favor the solution that results in less errors thrown. I think it's intuitive that private methods aren't seen by Motif.